ActivatePubKey/__tests__/handler.test.ts

```diff
@@ -0,0 +1,260 @@
+import { PubKeyStatusEnum } from "../../generated/definitions/internal/PubKeyStatus";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import { ActivatePubKeyHandler } from "../handler";
+import { BlobService, createBlobService } from "azure-storage";
+import { getPopDocumentReader } from "../../utils/readers";
+import { getAssertionWriter, getPopDocumentWriter } from "../../utils/writers";
+import { ActivatePubKeyPayload } from "../../generated/definitions/internal/ActivatePubKeyPayload";
+import {
+  retrievedLollipopKeysToApiActivatedPubKey,
+  RetrievedValidPopDocument
+} from "../../utils/lollipopKeys";
+import * as fn_commons from "@pagopa/io-functions-commons/dist/src/utils/azure_storage";
+import { getAllAssertionsRef } from "../../utils/lollipopKeys";
+import { JwkPubKeyHashAlgorithmEnum } from "../../generated/definitions/internal/JwkPubKeyHashAlgorithm";
+import {
+  aValidJwk,
+  aValidSha256AssertionRef,
+  aValidSha512AssertionRef,
+  toEncodedJwk
+} from "../../__mocks__/lollipopPubKey.mock";
+import { blobServiceMock } from "../../__mocks__/blobService.mock";
+
+const aFiscalCode = "SPNDNL80A13Y555X" as FiscalCode;
+
+const aValidRetrievedPopDocument: RetrievedLolliPopPubKeys = {
+  pubKey: toEncodedJwk(aValidJwk),
+  ttl: TTL_VALUE_AFTER_UPDATE,
+  assertionType: AssertionTypeEnum.SAML,
+  assertionRef: aValidSha256AssertionRef,
+  assertionFileName: `${aFiscalCode}-${aValidSha256AssertionRef}` as AssertionFileName,
+  status: PubKeyStatusEnum.VALID,
+  fiscalCode: aFiscalCode,
+  expiredAt: new Date(),
+  id: "1" as NonEmptyString,
+  version: 0 as NonNegativeInteger,
+  _etag: "",
+  _rid: "",
+  _self: "",
+  _ts: 0
+};
+
+const aPendingRetrievedPopDocument: RetrievedLolliPopPubKeys = {
+  ...aValidRetrievedPopDocument,
+  status: PubKeyStatusEnum.PENDING
+};
+
+const aValidRetrievedPopDocumentWithMasterAlgo = {
+  ...aValidRetrievedPopDocument,
+  assertionRef: aValidSha512AssertionRef,
+  assertionFileName: `${aFiscalCode}-${aValidSha512AssertionRef}`
+} as RetrievedValidPopDocument;
+
+const aPendingRetrievedPopDocumentWithMasterAlgo = {
+  ...aPendingRetrievedPopDocument,
+  // to let this document match the master we must change the assertionRef
+  // to a sha512 one
+  assertionRef: aValidSha512AssertionRef,
+  assertionFileName: `${aFiscalCode}-${aValidSha512AssertionRef}`
+};
+
+const upsertBlobFromTextMock = jest.spyOn(fn_commons, "upsertBlobFromText");
+
+const findLastVersionByModelIdMock = jest
+  .fn()
+  .mockImplementation(() => TE.of(O.some({})));
+
+const upsertMock = jest.fn().mockImplementation(() => TE.of({}));
+
+const lollipopPubKeysModelMock = ({
+  findLastVersionByModelId: findLastVersionByModelIdMock,
+  upsert: upsertMock
+} as unknown) as LolliPOPKeysModel;
+
+const contextMock = {} as any;
+
+const LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME = "assertions" as NonEmptyString;
+
+describe("activatePubKey handler", () => {
```

Comment:
why separate errors from success into 2 different files instead of provide different describe here?

Reply:
we were working in parallel PRs with @gquadrati. those tests can now be collapsed in a single file TBH.

Reply:
done in https://github.com/pagopa/io-functions-lollipop/pull/13/commits/174ea38fa5c1d0b6c390e3517ce72eea2e9c57d4